spring-cloud-kubernetes-client-config/src/main/java/org/springframework/cloud/kubernetes/client/config/reload/KubernetesClientConfigReloadAutoConfiguration.java

```diff
@@ -112,6 +117,7 @@ public ConfigurationChangeDetector secretsPropertyChangePollingWatcher(ConfigRel
 	 * @return a bean that listen to configMap change events and fire a reload.
 	 */
 	@Bean
+	@ConditionalOnConfigMapsReloadEnabled
```

Comment:
I've created `ConditionalOnConfigMapsReloadEnabled` and `ConditionalOnSecretsReloadEnabled`.

Until now, we were creating the beans and in post construct were checking if `monitoring-secrets/monitoring-configMaps` is enabled. If it was not - we were basically doing nothing; but this begs the question - why bother creating the beans in such a case? 

As such, I created those two annotations and use them as conditionals on bean creation.